resources/components/ImageCard.vue

```diff
@@ -188,25 +197,52 @@ module.exports = {
 				imgTitle: this.imgTitle
 			} ).connect( this, { confirm: 'publishTags' } );
 
-			$( document.body ).append( windowManager.$element );
-			windowManager.addWindows( [ this.confirmTagsDialog ] );
-			windowManager.openWindow( this.confirmTagsDialog );
+			this.windowManager.addWindows( [ this.confirmTagsDialog ] );
+			this.windowManager.openWindow( this.confirmTagsDialog );
 		},
 
 		/**
 		 * Skip the image (remove it from the Vuex queue).
 		 */
 		onSkip: function () {
 			this.skipImage();
+		},
+
+		/**
+		 * Launch the "add custom tag" dialog (OOUI modal)
+		 */
+		launchCustomTagDialog: function () {
+			this.windowManager.openWindow( this.addCustomTagDialog );
 		}
-	} )
+	} ),
+
+	/**
+	 * We are still relying on OOUI for modals, so we need to set up the
+	 * WindowManager and dialog widgets when the component mounts.
+	 * Once MediaWiki exposes an appropriate DOM element for vue-based modals
+	 * to target, we can rewrite this functionality in Vue. This is an
+	 * exception to the general rule of "don't manipulate the DOM directly from
+	 * Vue".
+	 */
+	mounted: function () {
+		this.windowManager = new OO.ui.WindowManager();
+		this.addCustomTagDialog = new AddCustomTagDialog().connect( this, {
+			addCustomTag: 'addCustomTag'
+		} );
+
+		$( document.body ).append( this.windowManager.$element );
+		this.windowManager.addWindows( [ this.addCustomTagDialog ] );
+	}
 };
 </script>
 
 <style lang="less">
 @import 'mediawiki.mixins';
+@import '../../lib/wikimedia-ui-base.less';
```

Comment:
@egardner There should be no need to import this here since it's imported in `style-variables.less`